internal/controller/config.go

```diff
@@ -17,28 +17,29 @@ package controller
 import (
 	"path/filepath"
 
-	"github.com/linkall-labs/vanus/internal/controller/snowflake"
-
 	embedetcd "github.com/linkall-labs/embed-etcd"
 	"github.com/linkall-labs/vanus/internal/controller/eventbus"
+	"github.com/linkall-labs/vanus/internal/controller/snowflake"
 	"github.com/linkall-labs/vanus/internal/controller/trigger"
 	"github.com/linkall-labs/vanus/internal/primitive"
+	"github.com/linkall-labs/vanus/observability"
 )
 
 type Config struct {
-	NodeID               uint16            `yaml:"node_id"`
-	Name                 string            `yaml:"name"`
-	IP                   string            `yaml:"ip"`
-	Port                 int               `yaml:"port"`
-	GRPCReflectionEnable bool              `yaml:"grpc_reflection_enable"`
-	EtcdEndpoints        []string          `yaml:"etcd"`
-	DataDir              string            `yaml:"data_dir"`
-	MetadataConfig       MetadataConfig    `yaml:"metadata"`
-	EtcdConfig           embedetcd.Config  `yaml:"embed_etcd"`
-	Topology             map[string]string `yaml:"topology"`
-	Replicas             uint              `yaml:"replicas"`
-	SecretEncryptionSalt string            `yaml:"secret_encryption_salt"`
-	SegmentCapacity      int64             `yaml:"segment_capacity"`
+	NodeID               uint16               `yaml:"node_id"`
+	Name                 string               `yaml:"name"`
+	IP                   string               `yaml:"ip"`
+	Port                 int                  `yaml:"port"`
+	GRPCReflectionEnable bool                 `yaml:"grpc_reflection_enable"`
+	EtcdEndpoints        []string             `yaml:"etcd"`
+	DataDir              string               `yaml:"data_dir"`
+	MetadataConfig       MetadataConfig       `yaml:"metadata"`
+	EtcdConfig           embedetcd.Config     `yaml:"embed_etcd"`
+	Topology             map[string]string    `yaml:"topology"`
+	Replicas             uint                 `yaml:"replicas"`
+	SecretEncryptionSalt string               `yaml:"secret_encryption_salt"`
+	SegmentCapacity      int64                `yaml:"segment_capacity"`
+	Observability        observability.Config `yaml:"observability"`
```

Comment:
do you think we need a struct CommonConfig which all component needs, such as Port, ControllerAddrs, Observability, and LogConfig

Reply:
Yes, I created a https://github.com/linkall-labs/vanus/issues/343 to track this